### DISCUSSION TITLE
Making REST API calls with a Cli generated app

### DISCUSSION THREAD

**Discussion by SaiSa22** [Shopify/shopify-app-template-node#1121] [Q&A] [answered]
I would appreciate the group's guidance on how to best leverage the node template cli generated app to make REST API calls. There are a lot of built in helpers that I would like to use. 

To start I wanted to expand on the "ProductsCard" component and modify the code below to point to a different URL. I am unable to use "useAppQuery" and would be great to get some help. 

  const {
    data,
    refetch: refetchProductCount,
    isLoading: isLoadingCount,
    isRefetching: isRefetchingCount,
  } = useAppQuery({
    url: "/api/products/count",
    reactQueryOptions: {
      onSuccess: () => {
        setIsLoading(false);
      },
    },
  }); 

I wanted to fetch the total number of customers which should be simple end point change to "/admin/api/customers/count" however this generates this error: "Failed to load resource: the server responded with a status of 500 ()"

**Comment by fonya2014** [comment]
Hi @SaiSa22  ! As I can see here: https://shopify.dev/api/admin-rest/2022-04/resources/customer
You probably need to add the API version:
![image](https://user-images.githubusercontent.com/64004218/196166410-dfb5c65c-e555-4b86-8497-df2bb9033297.png)

Try 2022-07 instead of 2022-04

**Reply by SaiSa22** [reply]
unfortunately I tried that already, and it does not work. I still get the same error. Any ideas what else I should try? This should be a pretty simple change, I am not sure why I am running into all sorts of issues.

**Comment by fonya2014** [comment]
Try to remove /admin/:
/api/customers/count
or
/2022-07/api/customers/count

**Reply by SaiSa22** [reply]
Thank you again for the suggestion. I tried those, and many other permutations of the url. Same issue. I tried with and without the version, with and without admin.

**Comment by fonya2014** [comment, marked as answer]
Hey @SaiSa22 ! You need to make changes in index.js on line 104:

```
app.get("/api/products/count", async (req, res) => {
    const session = await Shopify.Utils.loadCurrentSession(
      req,
      res,
      app.get("use-online-tokens")
    );
    /*const { Product } = await import(
      `@shopify/shopify-api/dist/rest-resources/${Shopify.Context.API_VERSION}/index.js`
    );*/

    const { Customer } = await import(
      `@shopify/shopify-api/dist/rest-resources/${Shopify.Context.API_VERSION}/index.js`
    );

    const countData = await Customer.count({ session });
    res.status(200).send(countData);
  });
```

You need to replace Product with Customer.

You don't need to change the URL in ProductsCard.jsx.

The URL in ProductsCard.jsx and in index.js:

`app.get("/api/products/count", async (req, res) => {`

must be equal to make it work, this is "Routing".

**Reply by SaiSa22** [reply]
That did it!! Thank you so much for the help. Really appreciate the support!

**Reply by fonya2014** [reply]
Any time!!!
